### ISSUE TITLE
Allow generating dependent protos using in different languages

### ISSUE THREAD

**Issue by Egorand** [square/wire#731]
kotlin.proto:
```kotlin
message Kotlin {
  optional string foo = 1;
}
```

java.proto:
```java
import "kotlin.proto";

message JavaMessage {
  optional Kotlin kotlin = 1;
}
```

1) Running the compiler on `java.proto` with `--java_out` generates Java files for both protos, hence subsequent invocation on `kotlin.proto` with `--kotlin_out` generates a duplicate `Kotlin` file.
2) Running the compiler on `kotlin.proto` with `--kotlin_out` generates Kotlin file, but subsequent invocation on `java.proto` with `--java_out` generates a duplicate `Kotlin.java` file.

**Comment by Egorand** [comment]
@mc-lovin any idea how we can overcome this?

**Comment by Egorand** [comment]
Yeah, that was the plan. Guess it's time to start working on this.

**Comment by mc-lovin** [comment]
I think whitelist makes sense.

if the flag for `java_interop` is set we can scan the whitelist and generate kotlin **protos** only for entries present in the whitelist.

**Comment by swankjesse** [comment]
Could we use profiles for this?

**Comment by Egorand** [comment]
@swankjesse yes!

**Comment by Egorand** [comment]
This should be covered by #742
